post_office/contrib/sendgrid/views.py

```diff
@@ -0,0 +1,296 @@
+}
+
+
+def serialize_email(email):
+    return (
+        None
+        if email is None
+        else {
+            'message_id': email.message_id,
+            'to': email.to,
+            'sent': email.created,
+            'subject': email.subject,
+        }
+    )
+
+
+class SendgridWebhookHandler(BaseWebhookHandler):
+    def verify_webhook(self, request: HttpRequest, *args, **kwargs) -> bool:
+        verified = False
+        try:
+            verified = check_sendgrid_signature(request)
+        except Exception:
+            verified = False
+        finally:
+            return verified
+
+    def post(self, request: HttpRequest, *args, **kwargs) -> HttpResponse:
+        try:
+            # Data is coming from JSON, where the root object isn't necessarily
+            # guaranteed to be a list.
+            # We reverse this because the list appears to be ordered reverse-chronologically
+            for payload in reversed(json.loads(request.body)):
+                logger.debug(payload)
+
+                # If we can't pick out the exact email, don't try to do anything
+                # This can happen if we use django-post_office and generate Emails and Logs
+                # before we configure and use this webhook handler
+                email = Email.objects.filter(message_id=payload.get('smtp-id', None)).first()
+                event = SENDGRID_STATUS_TO_EVENT.get(payload['event'], payload['event'])
+
+                self.handle_event(request, event, payload, *args, email=email, **kwargs)
+
+        except Exception as e:
+            logger.error(e)
+            # Raise an exception here so Django returns a 500 and emails the admins
+            # that something is wrong
+            raise Exception(
+                'Error when processing Sendgrid Events webhook. The Sendgrid Events Webhook API may have changed.'
+            )
+        return HttpResponse('ok')
+
+    def handle_event(
+        self, request: HttpRequest, event: Event, payload: dict[str, Any], *args, email: Email | None = None, **kwargs
+    ) -> None:
+        if event in [
+            Event.ACCEPTED,
+            Event.DELIVERED,
+            Event.DEFERRED,
+            Event.HARD_BOUNCE,
+            Event.SOFT_BOUNCE,
+            Event.REJECTED,
+        ]:
+            return self.log_deliverability_event(request, event, payload, *args, email=email, **kwargs)
+        elif event in [
+            Event.OPEN,
+            Event.CLICK,
+            Event.SPAM_COMPLAINT,
+            Event.UNSUBSCRIBE,
+            Event.RESUBSCRIBE,
+        ]:
+            return self.log_engagement_event(request, event, payload, *args, email=email, **kwargs)
+        elif event == Event.ACCOUNT_SUSPENDED:
+            return self.account_suspended(request, event, payload, *args, email=email, **kwargs)
+        return self.unrecognized_event(request, event, payload, *args, email=email, **kwargs)
+
+    def unrecognized_event(
```

Comment:
Please remove `unrecognized_event`, `log_deliverability_event` and `account_suspended`. We can cite this as an example of how to use this webhook in the docs.